File: 20240910/jiyeon/정복자.py
def find(c):
    if(c != parent[c]):
        parent[c] = find(parent[c])
    return parent[c]

def union(a, b):
    aa = find(a)
    bb = find(b)
    parent[max(aa, bb)] = min(aa, bb)

n, m, t = map(int, input().split())
graph = [list(map(int, input().split())) for _ in range(m)]
parent = [i for i in range(n+1)]
money = 0
edgeCnt = 0

graph.sort(key=lambda x : x[2])

for a, b, c in graph:
    if(find(a) != find(b)):
        union(a, b)
        money += c
        edgeCnt += 1

extraMoney = 0
for i in range(n-2):
    extraMoney += (i+1)*t

print(money + extraMoney)

File: 20240910/haeun/정복자.js
const fs = require("fs");
const [[N, M, T], ...input] = fs
  .readFileSync(0)
  .toString()
  .trim()
  .split("\n")
  .map((line) => line.split(" ").map(Number));
const parent = Array.from({ length: N }, (_, i) => i);
let sum = 0;
input.sort((a, b) => a[2] - b[2]);

for (let i = 0; i < M; i++) {
  const [from, to, cost] = input[i];
  const fa = find(from);
  const fb = find(to);

  if (fa != fb) {
    union(from, to);
    sum += cost;
  }
}

console.log(sum + (((N - 2) * (N - 2 + 1)) / 2) * T);

function find(a) {
  if (parent[a] !== a) parent[a] = find(parent[a]);
  return parent[a];
}

function union(a, b) {
  const fa = find(a);
  const fb = find(b);

  if (fa !== fb) parent[fa] = fb;
}
